lib/Conversion/ImportVerilog/Statement.cpp

```diff
@@ -0,0 +1,149 @@
+//===- Statement.cpp - Slang statement conversion--------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImportVerilogInternals.h"
+#include "slang/ast/ASTVisitor.h"
+#include "slang/ast/Symbol.h"
+#include "slang/ast/symbols/CompilationUnitSymbols.h"
+#include "slang/ast/symbols/InstanceSymbols.h"
+#include "slang/ast/symbols/VariableSymbols.h"
+#include "slang/ast/types/AllTypes.h"
+#include "slang/ast/types/Type.h"
+#include "slang/syntax/SyntaxVisitor.h"
+
+using namespace circt;
+using namespace ImportVerilog;
+
+void Context::convertExpression(const slang::ast::Expression *expression) {
+  auto loc = convertLocation(expression->sourceRange.start());
+  switch (expression->kind) {
+  case slang::ast::ExpressionKind::IntegerLiteral:
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::NamedValue:
+    // I think that needs to call another function, like visitExpr(...), which
+    // handles details about expressions.
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::UnaryOp:
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::BinaryOp:
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::Assignment:
+    assert(0 && "TODO");
+    break;
+  case slang::ast::ExpressionKind::Conversion:
+    assert(0 && "TODO");
+    break;
+    // Other cases need to be appended.
+  default:
+    mlir::emitError(loc, "unsupported ExpressionKind");
+    break;
+  }
+}
+
+void Context::convertStatement(const slang::ast::Statement *statement) {
+  auto loc = convertLocation(statement->sourceRange.start());
+  switch (statement->kind) {
+  case slang::ast::StatementKind::List:
+    for (auto it = statement->as<slang::ast::StatementList>().list.begin();
+         it != statement->as<slang::ast::StatementList>().list.end(); it++) {
+      convertStatement(*it);
```

Comment:
```suggestion
    for (auto *stmt : statement->as<slang::ast::StatementList>().list) {
      convertStatement(stmt);
```

Reply:
I learned it.  It is more concise and readable than the previous way.